msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.

**Describe the bug**
Azure automatically creates a resource-group for you. Mine is called "cloud-shell-storage-northeurope". 

**To Reproduce**
Click "Cloud Shell"
Choose a storage solution
mkdir <anything>
cd <anything>
az webapp up -n <app-name>

**Expected behavior**
I expected the 'resource_group_name' automatically generated by Azure not to fail it's own validation.

**Environment summary**
Azure CLI 2.0


**Additional context**
The command failed with an unexpected error. Here is the traceback:

Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.

@panchagnula Here's everything else being tried following the suggestions. Similar to others this is a new Azure account. And I'm using Free Trial. If you have a new account to have a try, there's a good chance to reproduce it. It's a very bad experience to start playing with Azure by following this quickstart-python guide. 

jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az extension list
[]
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az webapp up -n Jack_webapp_test1
The command failed with an unexpected error. Here is the traceback:

Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.

To open an issue, please run: 'az feedback'
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az webapp up -n Jack_webapp_test1 --debug
Command arguments: ['webapp', 'up', '-n', 'Jack_webapp_test1', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f71b1e258c8>, <function OutputProducer.on_global_arguments at 0x7f71b1966620>, <function CLIQuery.on_global_arguments at 0x7f71b198b0d0>]
Event: CommandInvoker.OnPreCommandTableCreate []
Installed command modules ['acr', 'acs', 'advisor', 'ams', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'interactive', 'iot', 'iotcentral', 'keyvault', 'kusto', 'lab', 'maps', 'monitor', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'reservations', 'resource', 'role', 'search', 'security', 'servicebus', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'vm']
Loaded module 'acr' in 0.004 seconds.
Loaded module 'acs' in 0.003 seconds.
Loaded module 'advisor' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'ams' in 0.016 seconds.
Loaded module 'appservice' in 0.007 seconds.
Loaded module 'backup' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'batch' in 0.008 seconds.
Loaded module 'batchai' in 0.003 seconds.
Loaded module 'billing' in 0.002 seconds.
Loaded module 'botservice' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'cdn' in 0.002 seconds.
Loaded module 'cloud' in 0.001 seconds.
Loaded module 'cognitiveservices' in 0.001 seconds.
Loaded module 'configure' in 0.001 seconds.
Loaded module 'consumption' in 0.002 seconds.
Loaded module 'container' in 0.002 seconds.
Loaded module 'cosmosdb' in 0.003 seconds.
Loaded module 'dla' in 0.004 seconds.
Loaded module 'dls' in 0.003 seconds.
Loaded module 'dms' in 0.002 seconds.
Loaded module 'eventgrid' in 0.002 seconds.
Loaded module 'eventhubs' in 0.003 seconds.
Loaded module 'extension' in 0.001 seconds.
Loaded module 'feedback' in 0.001 seconds.
Loaded module 'find' in 0.001 seconds.
Loaded module 'hdinsight' in 0.002 seconds.
Loaded module 'interactive' in 0.000 seconds.
Loaded module 'iot' in 0.004 seconds.
Loaded module 'iotcentral' in 0.001 seconds.
Loaded module 'keyvault' in 0.005 seconds.
Loaded module 'kusto' in 0.002 seconds.
Loaded module 'lab' in 0.003 seconds.
Loaded module 'maps' in 0.001 seconds.
Loaded module 'monitor' in 0.004 seconds.
Loaded module 'network' in 0.028 seconds.
Loaded module 'policyinsights' in 0.002 seconds.
Loaded module 'privatedns' in 0.004 seconds.
Loaded module 'profile' in 0.001 seconds.
Loaded module 'rdbms' in 0.005 seconds.
Loaded module 'redis' in 0.002 seconds.
Loaded module 'relay' in 0.003 seconds.
Loaded module 'reservations' in 0.002 seconds.
Loaded module 'resource' in 0.007 seconds.
Loaded module 'role' in 0.004 seconds.
Loaded module 'search' in 0.001 seconds.
Loaded module 'security' in 0.002 seconds.
Loaded module 'servicebus' in 0.004 seconds.
Loaded module 'servicefabric' in 0.002 seconds.
Loaded module 'signalr' in 0.001 seconds.
Loaded module 'sql' in 0.006 seconds.
Loaded module 'sqlvm' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'storage' in 0.030 seconds.
Loaded module 'vm' in 0.013 seconds.
Loaded all modules in 0.222 seconds. (note: there's always an overhead with the first module loaded)
Extensions directory: '/home/jackhasaboat/.azure/cliextensions'
Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f71b16f2510>]
az_command_data_logger : command args: webapp up -n {} --debug
metadata file logging enabled - writing logs to '/home/jackhasaboat/.azure/commands'.
Event: CommandInvoker.OnPostCommandTableCreate [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f71b1695488>, <function register_ids_argument.<locals>.add_ids_arguments at 0x7f71b16bfea0>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x7f71af37d7b8>]
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f71b19666a8>, <function CLIQuery.handle_query_parameter at 0x7f71b198b158>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f71b16bff28>, <function handler at 0x7f71af264950>]
Getting management service client client_type=WebSiteManagementClient
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
Current cloud config:
AzureCloud
Getting management service client client_type=WebSiteManagementClient
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
msrest.async_paging : Paging async iterator protocol is not available for GeoRegionPaged
urllib3.connectionpool : Starting new HTTP connection (1): localhost:50342
urllib3.connectionpool : http://localhost:50342 "POST /oauth2/token HTTP/1.1" 200 2060
msrestazure.azure_active_directory : MSI: Retrieving a token from http://localhost:50342/oauth2/token, with payload {'resource': 'https://management.core.windows.net/'}
urllib3.connectionpool : Starting new HTTP connection (1): localhost:50342
urllib3.connectionpool : http://localhost:50342 "POST /oauth2/token HTTP/1.1" 200 2060
msrestazure.azure_active_directory : MSI: Retrieving a token from http://localhost:50342/oauth2/token, with payload {'resource': 'https://management.core.windows.net/'}
msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions?sku=P1V2&linuxWorkersEnabled=true&api-version=2018-02-01'
msrest.http_logger : Request method: 'GET'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'User-Agent': 'python/3.6.5 (Linux-4.15.0-1041-azure-x86_64-with-debian-stretch-sid) msrest/0.6.6 cloud-shell/1.0 msrest_azure/0.6.0 azure-mgmt-web/0.41.0 Azure-SDK-For-Python AZURECLI/2.0.62 cloud-shell/1.0'
msrest.http_logger : Request body:
msrest.http_logger : None
msrest.universal_http : Configuring redirects: allow=True, max=30
msrest.universal_http : Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http : Configuring proxies: ''
msrest.universal_http : Evaluate proxies against ENV settings: True
urllib3.connectionpool : Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool : https://management.azure.com:443 "GET /subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions?sku=P1V2&linuxWorkersEnabled=true&api-version=2018-02-01 HTTP/1.1" 200 None
msrest.http_logger : Response status: 200
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Transfer-Encoding': 'chunked'
msrest.http_logger :     'Content-Type': 'application/json'
msrest.http_logger :     'Content-Encoding': 'gzip'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'Vary': 'Accept-Encoding'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'x-ms-request-id': 'b9448816-c1ff-4367-90a0-b1e9ba3556af'
msrest.http_logger :     'Server': 'Microsoft-IIS/10.0'
msrest.http_logger :     'X-AspNet-Version': '4.0.30319'
msrest.http_logger :     'X-Powered-By': 'ASP.NET'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-reads': '11999'
msrest.http_logger :     'x-ms-correlation-request-id': 'b7e48905-fa04-4a3f-9467-1495e7c50054'
msrest.http_logger :     'x-ms-routing-request-id': 'SOUTHEASTASIA:20190418T065601Z:b7e48905-fa04-4a3f-9467-1495e7c50054'
msrest.http_logger :     'X-Content-Type-Options': 'nosniff'
msrest.http_logger :     'Date': 'Thu, 18 Apr 2019 06:56:00 GMT'
msrest.http_logger : Response content:
msrest.http_logger : {"value":[{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Central US","name":"Central US","type":"Microsoft.Web/geoRegions","properties":{"name":"Central US","description":null,"sortOrder":0,"displayName":"Central US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/North Europe","name":"North Europe","type":"Microsoft.Web/geoRegions","properties":{"name":"North Europe","description":null,"sortOrder":1,"displayName":"North Europe"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/West Europe","name":"West Europe","type":"Microsoft.Web/geoRegions","properties":{"name":"West Europe","description":null,"sortOrder":2,"displayName":"West Europe"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Southeast Asia","name":"Southeast Asia","type":"Microsoft.Web/geoRegions","properties":{"name":"Southeast Asia","description":null,"sortOrder":3,"displayName":"Southeast Asia"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/East Asia","name":"East Asia","type":"Microsoft.Web/geoRegions","properties":{"name":"East Asia","description":null,"sortOrder":4,"displayName":"East Asia"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/West US","name":"West US","type":"Microsoft.Web/geoRegions","properties":{"name":"West US","description":null,"sortOrder":5,"displayName":"West US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/East US","name":"East US","type":"Microsoft.Web/geoRegions","properties":{"name":"East US","description":null,"sortOrder":6,"displayName":"East US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Japan West","name":"Japan West","type":"Microsoft.Web/geoRegions","properties":{"name":"Japan West","description":null,"sortOrder":7,"displayName":"Japan West"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Japan East","name":"Japan East","type":"Microsoft.Web/geoRegions","properties":{"name":"Japan East","description":null,"sortOrder":8,"displayName":"Japan East"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/East US 2","name":"East US 2","type":"Microsoft.Web/geoRegions","properties":{"name":"East US 2","description":null,"sortOrder":9,"displayName":"East US 2"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/North Central US","name":"North Central US","type":"Microsoft.Web/geoRegions","properties":{"name":"North Central US","description":null,"sortOrder":10,"displayName":"North Central US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/South Central US","name":"South Central US","type":"Microsoft.Web/geoRegions","properties":{"name":"South Central US","description":null,"sortOrder":11,"displayName":"South Central US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Brazil South","name":"Brazil South","type":"Microsoft.Web/geoRegions","properties":{"name":"Brazil South","description":null,"sortOrder":12,"displayName":"Brazil South"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Australia East","name":"Australia East","type":"Microsoft.Web/geoRegions","properties":{"name":"Australia East","description":null,"sortOrder":13,"displayName":"Australia East"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Australia Southeast","name":"Australia Southeast","type":"Microsoft.Web/geoRegions","properties":{"name":"Australia Southeast","description":null,"sortOrder":14,"displayName":"Australia Southeast"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/North CentralUS (Stage)","name":"North Central US (Stage)","type":"Microsoft.Web/geoRegions","properties":{"name":"North Central US (Stage)","description":null,"sortOrder":2147483647,"displayName":"North Central US (Stage)"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Central India","name":"Central India","type":"Microsoft.Web/geoRegions","properties":{"name":"Central India","description":null,"sortOrder":2147483647,"displayName":"Central India"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/West India","name":"West India","type":"Microsoft.Web/geoRegions","properties":{"name":"West India","description":null,"sortOrder":2147483647,"displayName":"West India"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/South India","name":"SouthIndia","type":"Microsoft.Web/geoRegions","properties":{"name":"South India","description":null,"sortOrder":2147483647,"displayName":"South India"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Canada Central","name":"Canada Central","type":"Microsoft.Web/geoRegions","properties":{"name":"Canada Central","description":null,"sortOrder":2147483647,"displayName":"Canada Central"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Canada East","name":"Canada East","type":"Microsoft.Web/geoRegions","properties":{"name":"Canada East","description":null,"sortOrder":2147483647,"displayName":"Canada East"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/West Central US","name":"West Central US","type":"Microsoft.Web/geoRegions","properties":{"name":"West Central US","description":null,"sortOrder":2147483647,"displayName":"West Central US"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/West US 2","name":"West US 2","type":"Microsoft.Web/geoRegions","properties":{"name":"WestUS 2","description":null,"sortOrder":2147483647,"displayName":"West US 2"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/UK West","name":"UK West","type":"Microsoft.Web/geoRegions","properties":{"name":"UK West","description":null,"sortOrder":2147483647,"displayName":"UK West"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/UK South","name":"UK South","type":"Microsoft.Web/geoRegions","properties":{"name":"UK South","description":null,"sortOrder":2147483647,"displayName":"UK South"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Central US EUAP","name":"Central US EUAP","type":"Microsoft.Web/geoRegions","properties":{"name":"Central US EUAP","description":null,"sortOrder":2147483647,"displayName":"Central US EUAP"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Korea South","name":"Korea South","type":"Microsoft.Web/geoRegions","properties":{"name":"Korea South","description":null,"sortOrder":2147483647,"displayName":"Korea South"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/Korea Central","name":"Korea Central","type":"Microsoft.Web/geoRegions","properties":{"name":"Korea Central","description":null,"sortOrder":2147483647,"displayName":"Korea Central"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/France Central","name":"France Central","type":"Microsoft.Web/geoRegions","properties":{"name":"France Central","description":null,"sortOrder":2147483647,"displayName":"France Central"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/South Africa North","name":"South Africa North","type":"Microsoft.Web/geoRegions","properties":{"name":"South Africa North","description":null,"sortOrder":2147483647,"displayName":"South Africa North"}},{"id":"/subscriptions/4c35434d-1e98-4e3f-862f-f20eed0f056c/providers/Microsoft.Web/geoRegions/South Africa West","name":"South Africa West","type":"Microsoft.Web/geoRegions","properties":{"name":"South Africa West","description":null,"sortOrder":2147483647,"displayName":"South Africa West"}}],"nextLink":null,"id":null}
Getting management service client client_type=ResourceManagementClient
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
msrest.exceptions : Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
cli.azure.cli.core.util : The command failed with an unexpected error. Here is the traceback:

The command failed with an unexpected error. Here is the traceback:

cli.azure.cli.core.util : Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
cli.azure.cli.core.util :
To open an issue, please run: 'az feedback'

To open an issue, please run: 'az feedback'
az_command_data_logger : exit code: 1
telemetry.save : Save telemetry record of length 2419 in cache
telemetry.check : Negative: The /home/jackhasaboat/.azure/telemetry.txt was modified at 2019-04-18 06:50:39.290967, which in less than 600.000000s
command ran in 1.901 seconds.
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az configure --defaults
az configure: error: argument --defaults/-d: expected at least one argument
usage: az configure [-h] [--verbose] [--debug]
                    [--output {json,jsonc,table,tsv,yaml,none}]
                    [--query JMESPATH] [--defaults DEFAULTS [DEFAULTS ...]]
                    [--subscription _SUBSCRIPTION]
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az configure

Welcome to the Azure CLI! This command will guide you through logging in and setting some default values.

Your settings can be found at /home/jackhasaboat/.azure/config
Your current configuration is as follows:

[cloud]
name = AzureCloud

Environment variables:
AZURE_ACCESS_TOKEN_FILE = /tmp/accessTokens.json
AZURE_HTTP_USER_AGENT = cloud-shell/1.0

Do you wish to change your settings? (y/N): n

You're all set! Here are some commands to try:
 $ az login
 $ az vm create --help
 $ az feedback

jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az configure --defaults
az configure: error: argument --defaults/-d: expected at least one argument
usage: az configure [-h] [--verbose] [--debug]
                    [--output {json,jsonc,table,tsv,yaml,none}]
                    [--query JMESPATH] [--defaults DEFAULTS [DEFAULTS ...]]
                    [--subscription _SUBSCRIPTION]
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az --version
azure-cli                         2.0.62

acr                                2.2.4
acs                               2.3.21
advisor                            2.0.0
ams                                0.4.4
appservice                        0.2.17
backup                             1.2.3
batch                              4.0.0
batchai                            0.4.8
billing                            0.2.1
botservice                        0.1.10
cdn                                0.2.2
cloud                              2.1.1
cognitiveservices                  0.2.5
command-modules-nspkg               2.0.2
configure                         2.0.21
consumption                        0.4.2
container                         0.3.16
core                              2.0.62
cosmosdb                           0.2.9
dla                                0.2.5
dls                                0.1.8
dms                                0.1.3
eventgrid                          0.2.2
eventhubs                          0.3.4
extension                          0.2.5
feedback                           2.2.0
find                               0.3.2
hdinsight                          0.3.3
interactive                        0.4.3
iot                                0.3.7
iotcentral                         0.1.6
keyvault                          2.2.14
kusto                              0.2.2
lab                                0.1.6
maps                               0.3.4
monitor                           0.2.12
network                            2.3.6
nspkg                              3.0.3
policyinsights                     0.1.2
privatedns                         1.0.0
profile                            2.1.5
rdbms                              0.3.9
redis                              0.4.2
relay                              0.1.4
reservations                       0.4.2
resource                          2.1.13
role                               2.5.0
search                             0.1.1
security                           0.1.1
servicebus                         0.3.4
servicefabric                     0.1.16
signalr                            1.0.0
sql                                2.2.1
sqlvm                              0.1.1
storage                            2.4.0
telemetry                          1.0.2
vm                                2.2.18

Python location '/opt/az/bin/python3'
Extensions directory '/home/jackhasaboat/.azure/cliextensions'

Python (Linux) 3.6.5 (default, Apr  4 2019, 22:51:52)
[GCC 5.4.0 20160609]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az webapp up -n Jack_webapp_test2 -l 'west Europe'
The command failed with an unexpected error. Here is the traceback:

Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.

To open an issue, please run: 'az feedback'
jackhasaboat@Azure:~/quickstart/python-docs-hello-world$ az webapp up -n Jack_webapp_test3 --dryrun
The command failed with an unexpected error. Here is the traceback:

Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2313, in webapp_up
    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 282, in should_create_new_rg
    elif (_check_resource_group_exists(cmd, rg_name) and
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/_create_util.py", line 86, in _check_resource_group_exists
    return rcf.resource_groups.check_existence(rg_name)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/resource/resources/v2018_05_01/operations/resource_groups_operations.py", line 61, in check_existence
    'resourceGroupName': self._serialize.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 672, in validate
    raise ValidationError(key, name, value)
msrest.exceptions.ValidationError: Parameter 'resource_group_name' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'.

To open an issue, please run: 'az feedback'

--- a/src/command_modules/azure-cli-appservice/HISTORY.rst
+++ b/src/command_modules/azure-cli-appservice/HISTORY.rst
@@ -4,7 +4,6 @@ Release History
 ===============
 * functionapp: deprecate `az functionapp devops-build` command. Rename it to `az functionapp devops-pipeline`
 * appservice plan: az appservice plan --sku documentation updated to reflect the supported appserviceplans
-* webapp: az webapp up supports local configuration and sets the values of the arguments on the local configuration on running the command
 * webapp: az webapp up supports optional arguments resourcegroup & plan to override the defaults offered by the command
 * webapp: az webapp ssh handles 'AZURE_CLI_DISABLE_CONNECTION_VERIFICATION' environment variable
 * appserviceplan: az appserviceplan create support for Linux FREE sku

--- a/src/command_modules/azure-cli-appservice/HISTORY.rst
+++ b/src/command_modules/azure-cli-appservice/HISTORY.rst
@@ -3,6 +3,9 @@
 Release History
 ===============
 * functionapp: deprecate `az functionapp devops-build` command. Rename it to `az functionapp devops-pipeline`
+* appservice plan: az appservice plan --sku documentation updated to reflect the supported appserviceplans
+* webapp: az webapp up supports local configuration and sets the values of the arguments on the local configuration on running the command
+* webapp: az webapp up supports optional arguments resourcegroup & plan to override the defaults offered by the command
 * webapp: az webapp ssh handles 'AZURE_CLI_DISABLE_CONNECTION_VERIFICATION' environment variable
 * appserviceplan: az appserviceplan create support for Linux FREE sku
 * webapp: az webapp up now has a 30s sleep after setting SCM_DO_BUILD_DURING_DEPLOYMENT=true appsetting to handle kudu cold start

--- a/src/command_modules/azure-cli-appservice/HISTORY.rst
+++ b/src/command_modules/azure-cli-appservice/HISTORY.rst
@@ -3,6 +3,7 @@
 Release History
 ===============
 * functionapp: deprecate `az functionapp devops-build` command. Rename it to `az functionapp devops-pipeline`
+* webapp: Fixing getting the correct username for cloudshell which was causing az webapp up to fail
 * appservice plan: az appservice plan --sku documentation updated to reflect the supported appserviceplans
 * webapp: az webapp up supports optional arguments resourcegroup & plan to override the defaults offered by the command
 * webapp: az webapp ssh handles 'AZURE_CLI_DISABLE_CONNECTION_VERIFICATION' environment variable

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py
@@ -263,10 +263,8 @@ def set_location(cmd, sku, location):
     return location
 
 
-def should_create_new_rg(cmd, default_rg, rg_name, is_linux):
-    if (default_rg and _check_resource_group_exists(cmd, default_rg) and
-            _check_resource_group_supports_os(cmd, default_rg, is_linux)):
-        return False
+# check if the RG value to use already exists and follows the OS requirements or new RG to be created
+def should_create_new_rg(cmd, rg_name, is_linux):
     if (_check_resource_group_exists(cmd, rg_name) and
             _check_resource_group_supports_os(cmd, rg_name, is_linux)):
         return False

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py
@@ -39,8 +39,8 @@ def load_arguments(self, _):
     # pylint: disable=line-too-long
     # PARAMETER REGISTRATION
     name_arg_type = CLIArgumentType(options_list=['--name', '-n'], metavar='NAME')
-    sku_arg_type = CLIArgumentType(help='The pricing tiers, e.g., F1(Free), D1(Shared), B1(Basic Small), B2(Basic Medium), B3(Basic Large), S1(Standard Small), P1(Premium Small), P1V2(Premium V2 Small), PC2 (Premium Container Small), PC3 (Premium Container Medium), PC4 (Premium Container Large)',
-                                   arg_type=get_enum_type(['F1', 'FREE', 'D1', 'SHARED', 'B1', 'B2', 'B3', 'S1', 'S2', 'S3', 'P1', 'P2', 'P3', 'P1V2', 'P2V2', 'P3V2', 'PC2', 'PC3', 'PC4']))
+    sku_arg_type = CLIArgumentType(help='The pricing tiers, e.g., F1(Free), D1(Shared), B1(Basic Small), B2(Basic Medium), B3(Basic Large), S1(Standard Small), P1V2(Premium V2 Small), PC2 (Premium Container Small), PC3 (Premium Container Medium), PC4 (Premium Container Large)',
+                                   arg_type=get_enum_type(['F1', 'FREE', 'D1', 'SHARED', 'B1', 'B2', 'B3', 'S1', 'S2', 'S3', 'P1V2', 'P2V2', 'P3V2', 'PC2', 'PC3', 'PC4']))
     webapp_name_arg_type = CLIArgumentType(configured_default='web', options_list=['--name', '-n'], metavar='NAME',
                                            completer=get_resource_name_completion_list('Microsoft.Web/sites'), id_part='name',
                                            help="name of the web app. You can configure the default using 'az configure --defaults web=<name>'")
@@ -362,6 +362,10 @@ def load_arguments(self, _):
 
     with self.argument_context('webapp up') as c:
         c.argument('name', arg_type=webapp_name_arg_type)
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('plan', options_list=['--plan', '-p'], configured_default='appserviceplan',
+                   completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
+                   help="name of the appserviceplan associated with the webapp")
         c.argument('sku', arg_type=sku_arg_type)
         c.argument('dryrun', help="show summary of the create and deploy operation instead of executing it", default=False, action='store_true')
         c.argument('location', arg_type=get_location_type(self.cli_ctx))

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
@@ -2273,8 +2273,8 @@ def get_history_triggered_webjob(cmd, resource_group_name, name, webjob_name, sl
     return client.web_apps.list_triggered_web_job_history(resource_group_name, name, webjob_name)
 
 
-def webapp_up(cmd, name, resource_group_name=None, plan=None, location=None, sku=None, dryrun=False,
-              logs=False, launch_browser=False):  # pylint: disable=too-many-statements, too-many-branches
+def webapp_up(cmd, name, resource_group_name=None, plan=None,  # pylint: disable=too-many-statements, too-many-branches
+              location=None, sku=None, dryrun=False, logs=False, launch_browser=False):
     import os
     from azure.cli.core._profile import Profile
     client = web_client_factory(cmd.cli_ctx)

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
@@ -2273,7 +2273,8 @@ def get_history_triggered_webjob(cmd, resource_group_name, name, webjob_name, sl
     return client.web_apps.list_triggered_web_job_history(resource_group_name, name, webjob_name)
 
 
-def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, launch_browser=False):  # pylint: disable=too-many-statements, too-many-branches
+def webapp_up(cmd, name, resource_group_name=None, plan=None, location=None, sku=None, dryrun=False,
+              logs=False, launch_browser=False):  # pylint: disable=too-many-statements, too-many-branches
     import os
     from azure.cli.core._profile import Profile
     client = web_client_factory(cmd.cli_ctx)
@@ -2305,6 +2306,7 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
         if sku is None:
             sku = lang_details.get("default_sku")
         else:
+            logger.info("Found sku argument, skipping use default sku")
             sku = sku
         is_skip_build = language.lower() == STATIC_RUNTIME_NAME
         os_val = "Linux" if language.lower() == NODE_RUNTIME_NAME \
@@ -2320,12 +2322,22 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
     location = set_location(cmd, sku, location)
     loc_name = location.replace(" ", "").lower()
     is_linux = True if os_val == 'Linux' else False
-    asp = "{}_asp_{}_{}_0".format(user, os_val, loc_name)
-    rg_name = "{}_rg_{}_{}".format(user, os_val, loc_name)
-    # Resource group: check if default RG is set
-    default_rg = cmd.cli_ctx.config.get('defaults', 'group', fallback=None)
-    logger.info("default RG value is: %s", default_rg)
-    _create_new_rg = should_create_new_rg(cmd, default_rg, rg_name, is_linux)
+
+    if resource_group_name is None:
+        logger.info('Using default ResourceGroup value')
+        rg_name = "{}_rg_{}_{}".format(user, os_val, loc_name)
+    else:
+        logger.info("Found user input for ResourceGroup %s", resource_group_name)
+        rg_name = resource_group_name
+
+    if plan is None:
+        logger.info('Using default appserviceplan value')
+        asp = "{}_asp_{}_{}_0".format(user, os_val, loc_name)
+        _asp_generic = asp[:-len(asp.split("_")[4])]  # used to determine if a new ASP needs to be created
+    else:
+        asp = plan
+        _asp_generic = asp
+    _create_new_rg = should_create_new_rg(cmd, rg_name, is_linux)
     logger.info("Should create new RG %s", _create_new_rg)
     src_path = "{}".format(src_dir.replace("\\", "\\\\"))
     rg_str = "{}".format(rg_name)
@@ -2343,6 +2355,11 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
             """ % (name, asp, rg_str, full_sku, os_val, location, src_path,
                    detected_version, runtime_version)
     create_json = json.loads(dry_run_str)
+    cmd.cli_ctx.config.set_value('defaults', 'group', rg_name)
+    cmd.cli_ctx.config.set_value('defaults', 'sku', full_sku)
+    cmd.cli_ctx.config.set_value('defaults', 'plan', asp)
+    cmd.cli_ctx.config.set_value('defaults', 'location', location)
+    cmd.cli_ctx.config.set_value('defaults', 'web', name)
 
     if dryrun:
         logger.warning("Web app will be created with the below configuration,re-run command "
@@ -2357,8 +2374,8 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
         _create_new_asp = True
     else:
         logger.warning("Resource group '%s' already exists.", rg_name)
-        _asp_generic = asp[:-len(asp.split("_")[4])]
         # get all asp in the RG
+        logger.warning("Verifying if the plan with the same sku exists or should create a new plan")
         data = (list(filter(lambda x: _asp_generic in x.name,
                             client.app_service_plans.list_by_resource_group(rg_name))))
         data_sorted = (sorted(data, key=lambda x: x.name))
@@ -2375,8 +2392,12 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
             # from the sorted data pick the last one & check if a new ASP needs to be created
             # based on SKU or not
             _plan_info = data_sorted[num_asps - 1]
-            _asp_num = int(_plan_info.name.split('_')[4]) + 1
-            asp = "{}_asp_{}_{}_{}".format(user, os_val, loc_name, _asp_num)
+            if plan is None:
+                _asp_num = int(_plan_info.name.split('_')[4]) + 1
+                asp = "{}_asp_{}_{}_{}".format(user, os_val, loc_name, _asp_num)
+            else:
+                asp = plan
+
     # create new ASP if an existing one cannot be used
     if _create_new_asp:
         logger.warning("Creating App service plan '%s' ...", asp)
@@ -2391,6 +2412,7 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
         logger.warning("App service plan '%s' already exists.", asp)
         _show_too_many_apps_warn = get_num_apps_in_asp(cmd, rg_name, asp) > 5
         _create_new_app = should_create_new_app(cmd, rg_name, name)
+    cmd.cli_ctx.config.set_value('defaults', 'plan', asp)
     # create the app
     if _create_new_app:
         logger.warning("Creating app '%s' ...", name)

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
@@ -2282,6 +2282,8 @@ def webapp_up(cmd, name, resource_group_name=None, plan=None,  # pylint: disable
     src_dir = os.getcwd()
     user = Profile().get_current_account_user()
     user = user.split('@', 1)[0]
+    if len(user.split('#', 1)) > 1:  # on cloudShell user is in format live.com#user@domain.com
+        user = user.split('#', 1)[1]
     logger.info("UserPrefix to use '%s'", user)
     # if dir is empty, show a message in dry run
     do_deployment = False if os.listdir(src_dir) == [] else True

--- a/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
+++ b/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py
@@ -2355,12 +2355,6 @@ def webapp_up(cmd, name, resource_group_name=None, plan=None,  # pylint: disable
             """ % (name, asp, rg_str, full_sku, os_val, location, src_path,
                    detected_version, runtime_version)
     create_json = json.loads(dry_run_str)
-    cmd.cli_ctx.config.set_value('defaults', 'group', rg_name)
-    cmd.cli_ctx.config.set_value('defaults', 'sku', full_sku)
-    cmd.cli_ctx.config.set_value('defaults', 'plan', asp)
-    cmd.cli_ctx.config.set_value('defaults', 'location', location)
-    cmd.cli_ctx.config.set_value('defaults', 'web', name)
-
     if dryrun:
         logger.warning("Web app will be created with the below configuration,re-run command "
                        "without the --dryrun flag to create & deploy a new app")
@@ -2412,7 +2406,6 @@ def webapp_up(cmd, name, resource_group_name=None, plan=None,  # pylint: disable
         logger.warning("App service plan '%s' already exists.", asp)
         _show_too_many_apps_warn = get_num_apps_in_asp(cmd, rg_name, asp) > 5
         _create_new_app = should_create_new_app(cmd, rg_name, name)
-    cmd.cli_ctx.config.set_value('defaults', 'plan', asp)
     # create the app
     if _create_new_app:
         logger.warning("Creating app '%s' ...", name)

